docs/details/signal.dox

```diff
@@ -211,32 +211,82 @@ respectively, then the possible batch operations are as follows.
 \defgroup signal_func_approx1 approx1
 \ingroup approx_mat
 
-approx1 interpolates data along the first dimensions.
-It has three options for the type of interpolation to perform:
-- Nearest neighbor  - \ref AF_INTERP_NEAREST
-- Linear interpolation  - \ref AF_INTERP_LINEAR
-- Bilinear interpolation - \ref AF_INTERP_BILINEAR
-- Cubic interpolation - \ref AF_INTERP_CUBIC
+approx1 performs interpolation on data along one dimension.
 
-Interpolation is performed assuming input data is equally spaced with indices
-in the range [0, n). The positions are sampled with respect to data at these
-locations.
+The input array, `in`, is comprised of values which lie on a range of
+indices defined by `idx_start` and `idx_step` [idx_start:idx_step:N]
+along dimension `interp_dim`.
```

Comment:
Like you did below for `off_grid`, we should also explain that the API of approx1 that doesn't take `idx_start` and `idx_step` assumes them to be of so and so values.
Similar, explanation for `interp_dim` is also needed.